app/routes/_layout+/tags.$tagId_.edit.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react'
+import type { LoaderArgs, V2_MetaFunction } from "@remix-run/node";
+import { json, redirect } from "@remix-run/node";
+import { Form, useLoaderData, useNavigation } from "@remix-run/react";
+import { parseFormAny, useZorm } from "react-zorm";
+import { z } from "zod";
+import Input from "~/components/forms/input";
+
+import { Button } from "~/components/shared/button";
+
+import { requireAuthSession, commitAuthSession } from "~/modules/auth";
+import { getTag, updateTag } from "~/modules/tag";
+import { assertIsPost, getRequiredParam, isFormProcessing } from "~/utils";
+import { appendToMetaTitle } from "~/utils/append-to-meta-title";
+import { sendNotification } from "~/utils/emitter/send-notification.server";
+
+export const UpdateTagFormSchema = z.object({
+  name: z.string().min(3, "Name is required"),
+  description: z.string(),
+});
+
+const title = "Edit Tag";
+
+export async function loader({ request, params }: LoaderArgs) {
+  await requireAuthSession(request);
+
+  const id = getRequiredParam(params, "tagId");
+  const tag = await getTag({ id })
+
+  const header = {
+    title,
+  };
+
+  return json({ header, tag });
+}
+
+export const meta: V2_MetaFunction<typeof loader> = ({ data }) => [
+  { title: data ? appendToMetaTitle(data.header.title) : "" },
+];
+
+export async function action({ request, params }: LoaderArgs) {
+  const authSession = await requireAuthSession(request);
+  assertIsPost(request);
+  const formData = await request.formData();
+  const result = await UpdateTagFormSchema.safeParseAsync(parseFormAny(formData));
+
+  const id = getRequiredParam(params, "tagId");
+
+  if (!result.success) {
+    return json(
+      {
+        errors: result.error,
+      },
+      { status: 400 }
+    );
+  }
+
+  await updateTag({
+    ...result.data,
+    id
+  });
+
+  sendNotification({
+    title: "Tag Updated",
+    message: "Your tag has been updated successfully",
+    icon: { name: "success", variant: "success" },
+    senderId: authSession.userId,
+  });
+
+  return redirect(`/tags`, {
+    headers: {
+      "Set-Cookie": await commitAuthSession(request, { authSession }),
+    },
+  });
+}
+
+export default function EditTag() {
+  const zo = useZorm("NewQuestionWizardScreen", UpdateTagFormSchema);
+  const navigation = useNavigation();
+  const disabled = isFormProcessing(navigation.state);
+  const { tag } = useLoaderData();
+
+  const [formData, setFormData] = useState({
+    name: tag.name, 
+    description: tag.description, 
+  })
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, field: string) => {
```

Comment:
I see this is the same we use for the categories edit. Can you please abstract it in a file in the `utils` folder and export it to use it for both places? Then we also have it for use in different components when needed.